cucumber-tsflow-specs/features/hooks.feature

```diff
@@ -282,3 +282,295 @@ Feature: Support for Cucumber hooks
         Then it passes
         And the hook "setup environment" was executed on scenario "example"
         And the hook "tear down environment" was executed on scenario "example"
+
+    Scenario: Binding a before step hook
+        Given a file named "features/a.feature" with:
+            """feature
+            Feature: Feature
+                Scenario: example
+                    Given a step
+            """
+        And a file named "step_definitions/steps.ts" with:
+            """ts
+            import {binding, given, beforeStep} from 'cucumber-tsflow';
+
+            @binding()
+            class Steps {
+                private state = 'hook has not executed';
+
+                @beforeStep()
+                public hook() {
+                    this.state = 'hook has executed';
+                }
+
+                @given("a step")
+                public given() {
+                    console.log(this.state);
+                }
+            }
+
+            export = Steps;
+            """
+        When I run cucumber-js
+        Then it passes
+        And the output does not contain "hook has not executed"
+        And the output contains "hook has executed"
+
+    Scenario: Binding a before step hook with multiple steps
+        Given a file named "features/a.feature" with:
+            """feature
+            Feature: Feature
+                Scenario: example
+                    Given a step
+                    And another step
+            """
+        And a file named "step_definitions/steps.ts" with:
+            """ts
+            import {binding, given, beforeStep, when} from 'cucumber-tsflow';
+
+            @binding()
+            class Steps {
+                private state = 'hook has not executed';
+
+                @beforeStep()
+                public hook() {
+                    this.state = 'hook has executed';
+                }
+
+                @given("a step")
+                public given() {
+                    console.log(this.state);
+                }
+
+                @when("another step")
+                public when() {
+                    console.log(this.state);
+                }
+            }
```

Comment:
This test does not validate the desired behavior that the hook runs before _each_ step. If it gets wired incorrectly internally to `Before,` this test won't detect it.

I think logging each function call would make the test clearer:
```suggestion
            class Steps {
                private counter = 0;

                @beforeStep()
                public hook() {
                    console.log(`${this.counter++} execution: beforeStep`);
                }

                @given("a step")
                public given() {
                    console.log(`${this.counter++} execution: given`);
                }

                @when("another step")
                public when() {
                    console.log(`${this.counter++} execution: when`);
                }
            }
```

Checking the logs for the entire scenario will show when each function is called, and the state is shared between them (called on the same instance).